src/Microsoft.ML.LightGBM.StaticPipe/LightGbmStaticExtensions.cs

```diff
@@ -139,18 +213,59 @@ public static class LightGbmStaticExtensions
             int? minDataPerLeaf = null,
             double? learningRate = null,
-            int numBoostRound = LightGbmArguments.Defaults.NumBoostRound,
-            Action<LightGbmArguments> advancedSettings = null,
+            int numBoostRound = Options.Defaults.NumBoostRound,
             Action<LightGbmRankingModelParameters> onFit = null)
         {
-            CheckUserValues(label, features, weights, numLeaves, minDataPerLeaf, learningRate, numBoostRound, advancedSettings, onFit);
+            CheckUserValues(label, features, weights, numLeaves, minDataPerLeaf, learningRate, numBoostRound, onFit);
             Contracts.CheckValue(groupId, nameof(groupId));
 
             var rec = new TrainerEstimatorReconciler.Ranker<TVal>(
                (env, labelName, featuresName, groupIdName, weightsName) =>
                {
                    var trainer = new LightGbmRankingTrainer(env, labelName, featuresName, groupIdName, weightsName, numLeaves,
-                       minDataPerLeaf, learningRate, numBoostRound, advancedSettings);
+                       minDataPerLeaf, learningRate, numBoostRound);
+
+                   if (onFit != null)
+                       return trainer.WithOnFitDelegate(trans => onFit(trans.Model));
+                   return trainer;
+               }, label, features, groupId, weights);
+
+            return rec.Score;
+        }
+
+        /// <summary>
+        /// Ranks a series of inputs based on their relevance, training a decision tree ranking model through the <see cref="LightGbmRankingTrainer"/>.
+        /// </summary>
+        /// <param name="catalog">The <see cref="RankingCatalog"/>.</param>
+        /// <param name="label">The label column.</param>
+        /// <param name="features">The features column.</param>
+        /// <param name="groupId">The groupId column.</param>
+        /// <param name="weights">The weights column.</param>
+        /// <param name="options">Algorithm advanced settings.</param>
+        /// <param name="onFit">A delegate that is called every time the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}.Fit(DataView{TInShape})"/> method is called on the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}"/> instance created out of this. This delegate will receive
+        /// the linear model that was trained. Note that this action cannot change the result in any way;
+        /// it is only a way for the caller to be informed about what was learnt.</param>
+        /// <returns>The set of output columns including in order the predicted binary classification score (which will range
+        /// from negative to positive infinity), the calibrated prediction (from 0 to 1), and the predicted label.</returns>
+        public static Scalar<float> LightGbm<TVal>(this RankingCatalog.RankingTrainers catalog,
+           Scalar<float> label, Vector<float> features, Key<uint, TVal> groupId, Scalar<float> weights,
```

Comment:
Ditto "`options` API" comment.

Reply:
these are static extensions  (details in other comment)

---
In reply to: [251534206](https://github.com/dotnet/machinelearning/pull/2261#discussion_r251534206) [](ancestors = 251534206)